src/parser/DeclParser.php

```diff
@@ -102,104 +106,72 @@ public function _fnStmt($is_method = false)
-            $body = iterator_to_array($this->_innerStmtList());
-            $this->parser->match(Tag::T_END);
+            $body = iterator_to_array($this->stmt_parser->_innerStmtList());
+            $this->reader->match(Tag::T_END);
         }
 
         return new FnStmt($signature, $body, $is_pub, $is_method, $is_short);
     }
 
-
-
-    public function _implDeclStmt()
+    public function _implStmt()
     {
-        // Structs are for properties
+        // Shapes are for properties
         // Classes are for methods
         $type = Tag::T_SHAPE;
-        $this->parser->match(Tag::T_IMPL);
-        $class_or_shape = $this->qualifiedName();
+        $this->reader->match(Tag::T_IMPL);
+        $class_or_shape = $this->name_parser->_qualifiedName();
         $class_for = null;
         // When it contains "for", it is being applied for a class
-        if ($this->parser->is(Tag::T_FOR)) {
+        if ($this->reader->is(Tag::T_FOR)) {
             $type = Tag::T_CLASS;
-            $this->parser->consume();
-            $class_for = $this->qualifiedName();
+            $this->reader->consume();
+            $class_for = $this->name_parser->_qualifiedName();
         }
 
         $body = new StmtList(iterator_to_array($this->_implStmtList()));
-        $this->parser->match(Tag::T_END);
+        $this->reader->match(Tag::T_END);
 
         return new ImplStmt($type, $class_or_shape, $class_for, $body);
     }
 
     public function _implStmtList()
     {
-        while ($this->parser->is(Tag::T_IDENT)) {
+        while ($this->reader->is(Tag::T_IDENT)) {
             yield $this->_fnStmt(/* implicit */ true);
         }
     }
 
     public function _moduleStmt()
     {
-        $this->parser->match(Tag::T_MODULE);
-        return new ModuleStmt($this->qualifiedName());
-    }
-
-    public function _openStmt()
-    {
-        $this->parser->match(Tag::T_OPEN);
-        $type = null;
-        if ($this->parser->is(Tag::T_CONST) || $this->parser->is(Tag::T_FN)) {
-            $type = $this->parser->consumeAndFetch();
-        }
-
-        $name = $this->parser->is('.') ? [$this->parser->consumeAndFetch()->getTag()] : [];
-        $name[] = $this->qualifiedName();
-        $alias = null;
-        $subprops = null;
-
-        if ($this->parser->is(Tag::T_AS)) {
-            $this->parser->consume();
-            $alias = $this->identifier();
-        } elseif ($this->parser->is('{')) {
-            $this->parser->consume();
-            $subprops[] = $this->identifier();
-
-            if ($this->parser->is(';')) {
-                do {
-                    $this->parser->match(';');
-                    $subprops[] = $this->identifier();
-                } while ($this->parser->is(';'));
-            }
-
-            $this->parser->match('}');
-        }
-
-        return new OpenStmt($name, $alias, $type, $subprops);
+        $this->reader->match(Tag::T_MODULE);
+        return new ModuleStmt($this->name_parser->_qualifiedName());
```

Comment:
`module` only defines the qualified name to identify the source, but I'm unsure if it will be kept here...